src/utilities/components.tsx

```diff
@@ -44,7 +44,9 @@ export function isElementOfType<P>(
     return false;
   }
 
-  const {type} = element;
+  const {type: defaultType} = element;
```

Comment:
Can we add a unit test?

Reply:
There is no existing unit tests for components. I would prefer to skip it for now.